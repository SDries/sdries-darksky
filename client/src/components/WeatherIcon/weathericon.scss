@import '../../assets/styles/mixins.scss';

.WeatherIcon {
	img {
		width: 100px;
		height: 100px;
		display: block;
	}

	&.big img {
		width: 150px;
		height: 150px;

		@include bp-small {
			width: 100px;
			height: 100px;
		}
	}

	&.huge img {
		width: 200px;
		height: 200px;

		@include bp-medium {
			width: 150px;
			height: 150px;
		}

		@include bp-small {
			width: 100px;
			height: 100px;
		}
	}
}
