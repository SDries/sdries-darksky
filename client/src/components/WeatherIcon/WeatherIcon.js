import React, { Component } from 'react';
import propTypes from 'prop-types';

import icons from '../../assets/images/icons.js';
import './weathericon.scss';

class WeatherIcon extends Component {
	constructor(props) {
		super(props);

		this.defaultIcon = 'clear-day';
	}

	isIconAvailable(iconName) {
		return icons.hasOwnProperty(iconName);
	}

	getSvgPath(icon) {
		return <img src={icons[icon]} alt={icon} />;
	}

	render() {
		if(!this.props.icon) return null;
		let classNames = ['WeatherIcon'];
		if(this.props.big) classNames.push('big');
		if(this.props.huge) classNames.push('huge');

		return <div className={classNames.join(' ')}>
		{this.getSvgPath(this.props.icon)}
		</div>;
	}
}

WeatherIcon.propTypes = {
	icon: propTypes.string,
	big: propTypes.bool,
	huge: propTypes.bool,
}

export default WeatherIcon;
