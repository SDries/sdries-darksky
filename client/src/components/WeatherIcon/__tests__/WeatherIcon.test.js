import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import WeatherIcon from '../WeatherIcon';

describe('react-weathericon', () => {
  test('renders nothing without icon name', () => {
    const iconComponent = mount((<WeatherIcon />));
    expect(toJson(iconComponent)).toMatchSnapshot();
  });

  test('renders icon when name is provided', () => {
    const iconComponent = mount((<WeatherIcon icon="clear-day"/>));
    expect(toJson(iconComponent)).toMatchSnapshot();
  });

  test('renders icon without "big" or "huge" className', () => {
    const iconComponent = mount((<WeatherIcon icon="iconName"/>));
    expect(iconComponent.find('.WeatherIcon').hasClass('big')).toBeFalsy();
    expect(iconComponent.find('.WeatherIcon').hasClass('huge')).toBeFalsy();
  });

  test('renders icon with "big" className', () => {
    const iconComponent = mount((<WeatherIcon icon="iconName" big/>));
    expect(iconComponent.find('.WeatherIcon').hasClass('big')).toBeTruthy();
  });

  test('renders icon with "huge" className', () => {
    const iconComponent = mount((<WeatherIcon icon="iconName" huge/>));
    expect(iconComponent.find('.WeatherIcon').hasClass('huge')).toBeTruthy();
  });
});
