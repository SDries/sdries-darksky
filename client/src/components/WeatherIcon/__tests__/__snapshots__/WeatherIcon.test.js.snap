// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-weathericon renders icon when name is provided 1`] = `
<WeatherIcon
  icon="clear-day"
>
  <div
    className="WeatherIcon"
  >
    <img
      alt="clear-day"
      src={
        Object {
          "ReactComponent": [Function],
          "__esModule": true,
          "default": "clear-day.svg",
        }
      }
    />
  </div>
</WeatherIcon>
`;

exports[`react-weathericon renders nothing without icon name 1`] = `<WeatherIcon />`;
